frontend/src/features/admin-form/settings/components/SecretKeyActivationModal.tsx

```diff
@@ -204,13 +257,20 @@ export const SecretKeyActivationModal = ({
                     })}
                     placeholder="Enter or upload your Secret Key to continue"
                   />
-                  <IconButton
+                  <Button
                     isDisabled={isLoading}
                     variant="outline"
-                    aria-label="Pass secret key from file"
-                    icon={<BiUpload />}
+                    rightIcon={<BiUpload />}
                     onClick={() => fileUploadRef.current?.click()}
-                  />
+                    onDragEnter={handleDragEnter}
+                    onDragLeave={handleDragLeave}
+                    onDragOver={handleDragOver}
+                    onDrop={handleDrop}
```

Comment:
Suggest putting this inside the current `Input` component and changing the placeholder when a file is being dragged instead of a new button. This would probably enable the best of both worlds -- the previous behaviour + this new drag and drop to read input behaviour.